src/som/compiler/KernanClient.java

```diff
@@ -354,12 +345,13 @@ public void run() {
       }
 
       // And parse the message itself
-      StringBuilder builder = new StringBuilder();
+      byte[] result = new byte[(int)len];
       for (int i = 0; i < len; i++) {
-        char c = (char) readByte();
-        builder.append(c);
+        result[i] = readByte();
       }
-      String message = builder.toString();
+      String message = null;
+      try { message = new String(result, "UTF8"); }
+      catch (java.io.UnsupportedEncodingException e) { assert false; }
```

Comment:
One more thing: this here is problematic. Swallowing exceptions like this is going to make for very unhappy users. For some reason, nothing happens if things go wrong. That's hard to diagnose.

I'd propose to wrap the exception into a RuntimeException instead. Then at least users will get a clue what's going wrong.

Reply:
That was the intention behind "assert false"; I beleive that said exception can never be thrown (since "UTF8" is a valid encoding); if I have made a mistake, the user should hopefully get a helpfull backtrace.

But I guess assertions can be turned off so your suggestion is probably better.

Reply:
Yeah, assertions aren't reliable unfortunately.